src/vs/editor/common/controller/cursorTypeOperations.ts

```diff
@@ -548,7 +547,26 @@ export class TypeOperations {
 		return autoClosingPair;
 	}
 
-	private static _isAutoClosingOpenCharType(config: CursorConfiguration, model: ITextModel, selections: Selection[], ch: string, insertOpenCharacter: boolean): StandardAutoClosingPairConditional | null {
+	private static _findSubAutoClosingPairClose(config: CursorConfiguration, autoClosingPair: StandardAutoClosingPairConditional): string {
+		// remove parts from largest sub-pair
+		let subPair: StandardAutoClosingPairConditional | null = null;
+		if (autoClosingPair && autoClosingPair.open.length > 1) {
+			let lastChar = autoClosingPair.close.slice(-1);
+			// get candidates with the same last character as close
+			let subPairCandidates = config.autoClosingPairs.autoClosingPairsCloseByEnd.get(lastChar) || [];
+			for (const subPairCandidate of subPairCandidates) {
+				if (subPair === null || subPairCandidate.open.length > subPair.open.length) {
+					if (autoClosingPair.open !== subPairCandidate.open && autoClosingPair.open.includes(subPairCandidate.open) && autoClosingPair.close.endsWith(subPairCandidate.close)) {
```

Comment:
Suggestion: move this into a function? 
`function is<Add Descriptive Name Here> (autoClosingPair, subPairCandidate) {
    return (autoClosingPair.open !== subPairCandidate.open && autoClosingPair.open.includes(subPairCandidate.open) && autoClosingPair.close.endsWith(subPairCandidate.close));
}
`

We could then use this to improve readability?

`If (is<Add Descriptive Name Here>(....) {
  subPair = subPairCandidate;
}
`

I only suggest the above, as I have no idea what the if statement is checking for.  The strongly named function would help future authors to understand the intent.

Reply:
Yes a descriptive name is a good idea. Do you think it's necessary to move into it's own function or just extract as a local variable?

It's basically checking that `subPairCandidate` actually is a subPair of `autoClosingPair`. That is to say, `autoClosingPair` strictly contains open and close and that `autoClosingPair` is only adding characters to the start of close and not the end. TBH I can't think of any good names for it other than "isActuallySubPair".

Reply:
isActuallySubPair is a good name.  Helps me and other future readers to understand what the intent is.